packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableTokenImage.tsx

```diff
@@ -0,0 +1,69 @@
+import { useMemo } from 'react'
+import EthereumLogoRoundLight from '@/images/EthereumLogoRoundLight.svg'
+import Image from 'next/image'
+
+import { useTokenLists } from '../../hooks/useTokenLists'
+import { ChainId } from '../../util/networks'
+import { MergedTransaction } from '../../state/app/state'
+import { orbitChains } from '../../util/orbitChainsList'
+import { AssetType } from '../../hooks/arbTokenBridge.types'
+
+export const TransactionsTableTokenImage = ({
+  tx
+}: {
+  tx: MergedTransaction
+}) => {
+  // we need to take token image from mainnet by symbol, some token images don't exists on other networks
+  const tokenLists = useTokenLists(ChainId.ArbitrumOne)
+
+  const allTokens = useMemo(() => {
+    return tokenLists.data?.flatMap(t => t.tokens) || []
+  }, [tokenLists])
+
+  const token = useMemo(() => {
+    return allTokens.find(
```

Comment:
We'd be going through a ton of tokens for mainnet, so better to construct a map and then access it by the key in O(1)